src/prefect/server/logs/messaging.py

```diff
@@ -0,0 +1,72 @@
+"""
+Log messaging for streaming logs through the messaging system.
+"""
+
+from contextlib import asynccontextmanager
+from typing import TYPE_CHECKING, AsyncGenerator, List
+
+from prefect.logging import get_logger
+from prefect.server.schemas.core import Log
+from prefect.server.utilities import messaging
+from prefect.settings.context import get_current_settings
+
+if TYPE_CHECKING:
+    import logging
+
+logger: "logging.Logger" = get_logger(__name__)
+
+
+@asynccontextmanager
+async def create_log_publisher() -> AsyncGenerator[messaging.Publisher, None]:
+    """
+    Creates a publisher for sending logs to the messaging system.
+
+    Returns:
+        A messaging publisher configured for the "logs" topic
+    """
+    async with messaging.create_publisher(topic="logs") as publisher:
+        yield publisher
+
+
+async def publish_log(log: Log) -> None:
+    """
+    Publishes a single log to the messaging system.
+
+    Args:
+        log: The log to publish
+    """
+    if not get_current_settings().server.logs.stream_publishing_enabled:
+        return
+
+    try:
+        async with create_log_publisher() as publisher:
+            await publisher.publish_data(
+                data=log.model_dump_json().encode(),
+                attributes={"log_id": str(log.id)} if log.id else {},
+            )
+    except Exception as e:
+        logger.warning(f"Failed to publish log to stream: {e}")
```

Comment:
Does this exception handler jive with what we're doing in events?  I'm not sure we should be swallowing errors here.